### PR TITLE
Keep topics in order

### LINKED ISSUE #213: Keep the order of help topics specified in _pkgdown.yml
It would be nice if this would be respected.

### PR DESCRIPTION
**Pull request by jranke**
Fixes #213 

Using this PR, `template` is now listed before `render_page` in docs/reference/index.html, i.e. topics are in the order that is specified in the YAML file. Everything else stays the same. The changes in docs/reference/build_reference.html are caused by a roxygen run before the last build_site(). The change in  docs/articles/pkgdown.html is caused by build_site(), updating the docs.

### CODE REVIEW

**Comment by jranke** [comment]
Would you prefer that I squash these commits?

**Comment by hadley** [comment]
I'll just squash on merge

**Comment by jranke** [comment]
@hadley Any blocking issues on this? I find it quite useful to keep the topics in the order specified in `_pkgdown.yml`.

File: R/build-reference-index.R [hunk 1]
```diff
@@ -95,6 +100,24 @@ match_topic <- function(topics, matches) {
...
 }
 
+# Returns the topic file name(s) for a set of aliases (matches)
+topic_files <- function(topics, matches) {
+
+  # Returns the topic file name(s) for a matcher function
+  find_topic_files <- function(matcher) {
+    topics %>%
+      purrr::keep(function(section_topic) any(purrr::map_lgl(section_topic, matcher))) %>%
+      names %>%
```
Comment (hadley) [inline R/build-reference-index.R]: `names()` always returns a vector, so I don't understand why you need an `unlist()` here?

File: R/build-reference-index.R [hunk 2]
```diff
@@ -95,6 +100,24 @@ match_topic <- function(topics, matches) {
...
+      unlist
+  }
+
+  matches %>%
+    purrr::map(topic_matcher) %>%
+    purrr::map(find_topic_files) %>%
+    unlist %>%
```
Comment (hadley) [inline R/build-reference-index.R]: Please use `()` after functions

File: R/build-reference-index.R [hunk 3]
```diff
@@ -95,6 +100,24 @@ match_topic <- function(topics, matches) {
     purrr::map_lgl(~ purrr::some(matchers, function(f) any(f(.))))
 }
 
+# Returns the topic file name(s) for a set of aliases (matches)
+topic_files <- function(topics, matches) {
+
+  # Returns the topic file name(s) for a matcher function
+  find_topic_files <- function(matcher) {
+    topics %>%
+      purrr::keep(function(section_topic) any(purrr::map_lgl(section_topic, matcher))) %>%
```
Comment (hadley) [inline R/build-reference-index.R]: It feels like some of this logic could be shared with `has_topic()`

**Comment by hadley** [comment]
Ended up completely rewriting how topic matching occurs so this was no longer relevant.

**Pull request closed without merging**